container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/engine_functions/engine_logs.go

```diff
@@ -0,0 +1,105 @@
+package engine_functions
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/shared_helpers"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/kubernetes_resource_collectors"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/label_key_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/engine"
+	"github.com/kurtosis-tech/stacktrace"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// GetEngineLogs dump pods, into the received output dirpath, for the current engine
+// it assumes that there is a single engine per cluster
+func GetEngineLogs(
+	ctx context.Context,
+	outputDirpath string,
+	kubernetesManager *kubernetes_manager.KubernetesManager,
+) error {
+	engineMatchLabels := getEngineMatchLabels()
+
+	engineGuidStrs := map[string]bool{}
+
+	// Namespaces
+	namespaces, err := kubernetes_resource_collectors.CollectMatchingNamespaces(
+		ctx,
+		kubernetesManager,
+		engineMatchLabels,
+		label_key_consts.IDKubernetesLabelKey.GetString(),
+		engineGuidStrs,
+	)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting engine namespaces")
+	}
+
+	if len(namespaces) == 0 {
+		return stacktrace.NewError("Didn't find the engine's namespace")
+	}
+	if len(namespaces) > 1 {
+		return stacktrace.NewError("Found more than one engine's namespace; this is a bug in Kurtosis")
+	}
+
+	for engineGuidStr, namespacesForId := range namespaces {
+		engineGuid := engine.EngineGUID(engineGuidStr)
+		if len(namespacesForId) > 1 {
+			return stacktrace.NewError(
+				"Expected at most one namespace to match engine GUID '%v', but got '%v'",
+				len(namespacesForId),
+				engineGuidStr,
+			)
+		}
+
+		engineNamespace := namespacesForId[0]
```

Comment:
are we sure that there's at least one name space? can the name spaces be a nil slice